src/policies/rate-limit-policy.ts

```diff
@@ -28,29 +31,51 @@ interface RateLimit {
 const rateLimitPolicy: Policy<RateLimit> = async (msg, opts = {}) => {
   const {
     interval = 60000,
+    ban_interval = 0,
     max = 10,
     whitelist = [],
-    databaseUrl = 'sqlite:///tmp/strfry-rate-limit-policy.sqlite3',
+    databaseUrl = "sqlite:///tmp/strfry-rate-limit-policy.sqlite3",
   } = opts;
 
-  if ((msg.sourceType === 'IP4' || msg.sourceType === 'IP6') && !whitelist.includes(msg.sourceInfo)) {
+  if (
+    (msg.sourceType === "IP4" || msg.sourceType === "IP6") &&
+    !whitelist.includes(msg.sourceInfo)
+  ) {
     const db = new Keydb(databaseUrl);
-    const count = await db.get<number>(msg.sourceInfo) || 0;
+    const count = (await db.get<number>(msg.sourceInfo)) || 0;
     await db.set(msg.sourceInfo, count + 1, interval);
 
     if (count >= max) {
+      if (ban_interval > 0) {
+        await db.set(`${msg.sourceInfo}-banned`, true, ban_interval);
+      }
+
+      return {
+        id: msg.event.id,
+        action: "reject",
+        msg: "rate-limited: too many requests",
+      };
+    }
+  }
+
+  if (ban_interval > 0) {
+    const db = new Keydb(databaseUrl);
+    const is_banned =
+      (await db.get<boolean>(`${msg.sourceInfo}-banned`)) ?? false;
+
+    if (is_banned) {
       return {
         id: msg.event.id,
-        action: 'reject',
-        msg: 'rate-limited: too many requests',
+        action: "shadowReject",
```

Comment:
Will this action be logged? If not we should definitely find another way to have traceability here.

Reply:
No, these don't get logged but I will add a `console.error` for pubkey, kind, and id.

Reply:
I added this type of lines:
`nosrelay-1  | Banned rate-limited IP 192.168.65.1. Pubkey: 79be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798, kind: 3, id 10b64b40007dfd9967b8dc23ae86196e059042cdd4520ccd1715f568d357d0c2.2024-09-20 23:54:59.504 ( 130.367s) [Websocket       ]INFO| [41] Disconnect from 192.168.65.1 (1006/Resource temporarily unavailable) UP: 81b (8.6% compressed) DN: 352b (30.1% compressed)`

Also more info more normal rate limits that were missing some